build-system/babel-plugins/babel-plugin-amd-custom-define/define-template.js

```diff
@@ -0,0 +1,52 @@
+// @ts-nocheck
+/* eslint-disable no-var */
+
+/* global __ARGS__ */
+/* global __MODULE_NAME__ */
+/* global __HAS_EXPORTS__ */
+/* global __ONLY_DEP__ */
+
+(function (deps, callback) {
```

Comment:
This is an implementation of `define` that can be DCE'd.

We could instead use esbuild's `inject` option to import a `define()` implementation. Since every binary bundles its own `define()`, using a custom babel plugin has the benefit that we can DCE it:

- `bento.js` only exports:

```js
function (S, Q) {
  self.BENTO = self.BENTO || {};
  var R = Q;
  (Q = function ($) {
    R.apply(null, arguments);
    var Z = 'bento',
      J = (self.BENTO[Z] = self.BENTO[Z] || {});
    for (J.m = $; J.c && J.c.length > 0; ) J.c.pop()($);
  }
  Q({});
})
```
- most extensions only import `bento`:

```js
function (s, j) {
  self.BENTO = self.BENTO || {};
  var d = self.BENTO.bento || {c: []};
  d.m ? j(d.m) : d.c.push(j);
})
```

- in case an extension imports multiple modules:

```js
function (f, E) {
  self.BENTO = self.BENTO || {};
  else
    Promise.all(
      f.map(function (b) {
        var h = self.BENTO[b] || {c: []};
        return h.m
          ? h.m
          : new Promise(function (z) {
              h.c.push(z);
            });
      })
    ).then(function (b) {
      E.apply(null, b);
    });
}
```

- in case an extension exports and imports, the largest implementation is included:

```js
function (l, x) {
  self.BENTO = self.BENTO || {};
  var z = x;
  if (
    ((x = function (y) {
      z.apply(null, arguments);
      var b = 'v0/bento-fit-text-1.0',
        h = (self.BENTO[b] = self.BENTO[b] || {});
      for (h.m = y; h.c && h.c.length > 0; ) h.c.pop()(y);
    }),
    !1)
  )
    var R;
  else
    Promise.all(
      l.map(function (y) {
        var b = self.BENTO[y] || {c: []};
        return b.m
          ? b.m
          : new Promise(function (h) {
              b.c.push(h);
            });
      })
    ).then(function (y) {
      x.apply(null, y);
    });
}
```